### DISCUSSION TITLE
Using `parser.models` forces `email-validator` dependency.

### DISCUSSION THREAD

**Discussion by peterschutt** [aws-powertools/powertools-lambda-python#1431] [Maintainers discussions]
I already use pydantic for domain modelling but before now haven't tried to use the parser models. 

I figured seeing I already have pydantic around I would have been good to go, but the package namespace imports in `utilities.parser.models` from `.ses` cause pydantic to throw an import error due to `from pydantic.networks import EmailStr` in `ses.py`.

This is the lib it forces to install is https://github.com/JoshData/python-email-validator - not a big library by any stretch, so the bloat effect would be on the small side.

Just figured I'd canvas opinion before opening an issue.

Cheers!

**Comment by heitorlessa** [comment]
Hey Pete, we've got a PR for this and an issue on this already.

https://github.com/awslabs/aws-lambda-powertools-python/pull/1259

**Reply by peterschutt** [reply]
My apologies! Thanks for the reference.

**Reply by heitorlessa** [reply]
No worries at all ;) we should get this addressed in the next release.

Pydantic / Parser and Event Source Data Classes are two features I'd like to rethink from the ground up in the future -- overlap, package size (modularisation), duplication, and dangers of raising ValidationError for event sources that don't have a schema IF we were to make Pydantic v2 the default experience (not extra)

**Reply by peterschutt** [reply]
Thanks:) I really should have searched first :facepalm:.

I haven't tested with pydantic v2 yet - the latest Talk Python to Me pod features it - sounds like big performance gains.

Perhaps the role that the dataclasses play today could be replaced by `TypedDict` definitions so users looking for more performance could choose to build around them and still get the type hints. Overlap between the pydantic models and dataclasses is definitely a thing that many wouldn't appreciate the need for, especially if the pydantic performance is to improve as much as it seems.

Thanks mate.

**Reply by peterschutt** [reply]
The topic in #1404 has me thinking about whether the pydantic models could be a totally separate library that powertools depends on so that they could maybe have a wider version support than the core library.

**Comment by heitorlessa** [comment]
That was the intention but we wouldn’t have the bandwidth to maintain both
- we might be able to next year.

For now, I think modularisation will be a great stepping stone. We’ll look
into that after 3.6 drop (v2).

**Reply by heitorlessa** [reply]
Update - this is no longer the case in V2. `email-validator` was dropped as a dependency. All dependencies being optional 

https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v2.0.0
